src/components/shared/Navbar.tsx

```diff
@@ -61,10 +65,91 @@ const NavbarUserSection = styled.div`
     }
 `;
 
+const userImage = css`
+    width: 40px;
+    height: 40px;
+    border-radius: 50%;
+`;
+
+const userBox = css`
+    display: flex;
+    align-items: center;
+    cursor: pointer;
+    position: relative;
+`;
+
+const modalContent = css`
+    background: #fff;
+    padding: 10px;
+    border-radius: 5px;
+    box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
+    position: absolute;
+    top: 150%;
+    right: 0;
+    min-width: 200px;
+    display: flex;
+    flex-direction: column;
+    z-index: 1000;
+`;
+
+const modalLink = css`
+    margin: 5px 0;
+    color: #1f2937;
+    text-decoration: none;
+    cursor: pointer;
+    &:hover {
+        color: #4b5563;
+    }
+`;
+
 function Navbar() {
-    // Grab user from user global state.
-    const user = useAppSelector((state) => state.setUser);
-    const userNameCapitalized = grabUserNameCapitalized(user);
+    const userId = useAppSelector((state) => state.user.userId);
+    const [user, setUser] = useState<User | null>();
```

Comment:
Here’s how to get the userId and query Firestore from an individual component (Part 1)

1. Access the userId from the Redux store using useAppSelector
2. Set up user state variable to use it inside the JSX